### PR TITLE
ROX-10695: CI: Fix for unbound rvm var

### PR DESCRIPTION
**Pull request by gavin-stackrox**
## Description

In Circle CI we see this error when running in machine environments:
`/opt/********/.rvm/scripts/functions/rvmrc_env: line 66: rvm_saved_env: unbound variable`
It seems to be a long standing bug in rvm (https://github.com/rvm/rvm/issues/4342). 
It is hard to know why this is occurring consistently now. The change to `set -u` scripts https://github.com/stackrox/stackrox/pull/520 was Feb 1. Perhaps the Circle CI machines were updated to an rvm with this bug.

## Checklist
- [x] Investigated and inspected CI test results

## Testing Performed

CI is sufficient

### CODE REVIEW

**Comment by gavin-stackrox** [comment]
Unfortunately the first hack just pushes the problem to the next unbound var `/opt/********/.rvm/scripts/functions/rvmrc_env: line 66: rvm_saved_env: unbound variable`

**Comment by gavin-stackrox** [comment]
Ideally we could just disable rvm but it is unknown as to what Circle CI dependencies exist on it.

**Comment by gavin-stackrox** [comment]
Applying `set -u` after sourcing other libs avoids the `rvm` issue.

File: scripts/ci/gke.sh [hunk 1]
```diff
@@ -1,16 +1,13 @@
 #!/usr/bin/env bash
 
-set -euo pipefail
-
 # A collection of GKE related reusable bash functions for CI
 
-set +u
 SCRIPTS_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")"/../.. && pwd)"
-set -u
-
```
Comment (gavin-stackrox) [inline scripts/ci/gke.sh:10]: Note this previous ugly pattern that was an attempt to workaround a different unbound var.

File: scripts/ci/gcp.sh [hunk 2]
```diff
@@ -1,15 +1,12 @@
 #!/usr/bin/env bash
 
-set -euo pipefail
```
Comment (vikin91) [inline scripts/ci/gcp.sh:3]: Could we maybe keep `set -eo pipefail`?

**Review by janisz — APPROVED** [review]

File: scripts/ci/gcp.sh (hunk 2 above)
Comment (gavin-stackrox) [reply scripts/ci/gcp.sh:3]: @vikin91 it is still there, just a few lines down. Why this fixes the rvm problem 🤷

**Pull request merged**